xsdata/formats/dataclass/serializers/xml.py

```diff
@@ -89,12 +89,37 @@ def render_node(self, obj, parent, namespaces: Namespaces) -> Element:
                     self.set_text(parent, value)
                 else:
                     self.render_sub_nodes(parent, value, var, namespaces)
-            else:
-                self.set_nil_attribute(parent, var, namespaces)
 
-        self.unset_nil_attribute(parent)
+        self.set_nil_attribute(parent, meta.nillable, namespaces)
         return parent
 
+    def next_value(self, vars: List[ClassVar], obj: Any):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/tefra/xsdata/pullRequest?prid=5244803)